main/src/main/scala/sbt/internals/parser/SbtParser.scala

```diff
@@ -130,7 +179,7 @@ private[sbt] case class SbtParser(file: File, lines: Seq[String]) extends Parsed
      * @return originalStatement or originalStatement with missing bracket
      */
     def parseStatementAgain(t: Tree, originalStatement: String): String = {
-      val statement = util.Try(toolbox.parse(originalStatement)) match {
+      val statement = util.Try(parse(originalStatement, fileName)) match {
         case util.Failure(th) =>
           val missingText = findMissingText(content, t.pos.end, t.pos.line, fileName, th)
           originalStatement + missingText
```

Comment:
I know this triple-parsing rigmarole bugs you, so I thought I'd leave you with the idea to ignore the end position of parsed statements, and instead  use the initial position of the subsequent statement (or the end position of the enclosing tree, if there is no subsequent statement) - 1.